### ISSUE TITLE
UBports installation failure - help wanted

### ISSUE THREAD

**Issue by PJK2011** [ubports/ubports-installer#593]
- Device: hammerhead
- Channel: installer-0.1.21-beta-x86_64.AppImage, 16.04/stable
- Build:
- Host: Kubuntu 18.04, adb and fastboot tested and working

### Steps to reproduce

Every time I run the installer, it stalls after flashing and installing the UBports Recovery Menu

### Expected behavior

Your 2018 video shows that at this point it should push a few more files to the phone and flash the system image.

### Actual behavior

The UBports Recovery menu comes up on the device, but in the installer, the circle spins and the text says:

"Waiting for device to enter recovery mode. Flashing recovery and boot images."

It will stay that way all night.

### Logfiles and additional information

Every time the log reads the same way:

Pug interceptor registered successfully
debug: Welcome to the party
debug: Killing all running adb servers...
debug: Using native platform tools!
debug: Running platform tool exec cmd adb kill-server
debug: 
debug: Running platform tool exec cmd adb -P 5038 kill-server
debug: 
debug: adb shell: echo 1
debug: Running platform tool exec cmd adb -P 5038 shell echo 1
debug: Successfully downloaded devices list
warn: Running latest stable version.
debug:  message=Command failed: adb -P 5038 shell echo 1
error: device unauthorized.
This adb server's $ADB_VENDOR_KEYS is not set
Try 'adb kill-server' if that seems wrong.
Otherwise check for a confirmation dialog on your device.
, stack=Error: Command failed: adb -P 5038 shell echo 1
error: device unauthorized.
This adb server's $ADB_VENDOR_KEYS is not set
Try 'adb kill-server' if that seems wrong.
Otherwise check for a confirmation dialog on your device.

    at ChildProcess.exithandler (child_process.js:282:12)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:921:16)
    at Socket.stream.socket.on (internal/child_process.js:348:11)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at Pipe._handle.close [as _onclose] (net.js:549:12), killed=false, code=1, signal=null, cmd=adb -P 5038 shell echo 1
    
(From here it looped back to the line:

debug: adb shell: echo 1

again. The loop will run all night. Each loop takes about 1 second.)

One time it got to a dialog box that I had never seen before. It contained this message:

Fastboot: Unknown error: <waiting for any device>
target reported max download size of 1073741824 bytes
sending 'recovery' (11828 KB)...
OKAY [  0.582s]
writing 'recovery'...
OKAY [  0.963s]
finished. total time: 1.545s
<waiting for any device>
target didn't report max-download size
sending 'boot' (12302 KB)...
FAILED (command write failed (Protocol error))
finished. total time: 0.000s

**Comment by PJK2011** [comment]
Problem solved:

1) Open a terminal at the ubports-installer-0.1.21-beta-x86_64.AppImage containing folder.

2) Give the newly downloaded file the necessary permissions with the command:

chmod u+x ubports-installer-0.1.21-beta-x86_64.AppImage

3) Start sudo session with the command:

sudo su

4) Run the installer with the command:

./ubports-installer-0.1.21-beta-x86_64.AppImage

I'll leave this issue open a few days in case anyone wants to add comments.

**Comment by JoTiZ** [comment]
I tried get the ubports-installer-0.1.21-beta-x86_64.AppImage working the way you described, but my hammerhead shows the old error behavior from above. Ending up in den UBports Recovery Menu and the Installer "waiting for device to enter recovery mode".

(Ubuntu 18.10 / ubports-installer-0.1.21-beta-x86_64.AppImage / hammerhead)

**Comment by PJK2011** [comment]
Did your terminal show that you are in the su mode after typing sudo su?

**Comment by JoTiZ** [comment]
To your question: Yes, after the sudo -s i was logged in as root, but the same behavior. 

The same problem occurred with Fedora 29, Windows 7 and Ubuntu 18.10.

**SOLUTION** (for my device): After some more research i installed lineageOS 14.1 on the hammerhead device and activated the developer mode again. After that the installation process with the ubports-installer-0.1.21-beta-x86_64.AppImage worked normally on Ubuntu 18.10. The last time I activated it the same way, but this time it worked correctly.

**Comment by PJK2011** [comment]
That's interesting. I figured my first failures were due to hammerhead being on LineageOS, so a UBPorts volunteer on Telegram recommended I flash back to stock. That didn't help either. I figured there were some permissions issues going on that caused the installer to halt. That's what gave me the idea to run the AppImage version as su. So there's something about switching back and forth between LineageOS and stock that is affecting the install. I also read a post that said the device isn't responding to the adb requests because app host does not have sufficient permission. Hence the su elevation, that for you, LineageOS liked, and for me, stock liked. I also had inconsistency with the Android device requesting the USB Debugging link due to the adb permissions.

**Comment by JoTiZ** [comment]
I just remembered that lineageOS 15.0 was installed on the hammerhead device when the problem occurred. I would have to test in which connection this was related to the problem.

**Comment by PJK2011** [comment]
That's interesting. I figured my first failures were due to hammerhead being on LineageOS, so a UBPorts volunteer on Telegram recommended I flash back to stock. That didn't help either. I figured there were some permissions issues going on that caused the installer to halt. That's what gave me the idea to run the AppImage version as su. So there's something about switching back and forth between LineageOS and stock that is affecting the install. I also read a post that said the device isn't responding to the adb requests because app host does not have sufficient permission. Hence the su elevation, that for you, LineageOS liked, and for me, stock liked. I also had inconsistency with the Android device requesting the USB Debugging link due to the adb permissions.
